users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+
+    @Nullable
+    User findUser(@Nullable String userName) {
+        if (userName == null) {
+            return null;
+        }
+        for (User user: userGetter()) {
+            if (userName.equals(user.name())) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    private boolean isSuperUser(String userName){
+        User user = findUser(userName);
+        return user != null && user.superuser();
+    }
+
+    private class PermissionVisitor extends AnalyzedStatementVisitor<SessionContext, Boolean> {
+
+        private final SelectStatementPermissionVisitor selectVisitor = new SelectStatementPermissionVisitor();
+
+        @Override
+        protected Boolean visitAnalyzedStatement(AnalyzedStatement analyzedStatement,
+                                                 SessionContext sessionContext) {
+            return true;
+        }
+
+        @Override
+        protected Boolean visitSelectStatement(SelectAnalyzedStatement analysis,
+                                               SessionContext sessionContext) {
+            if (!selectVisitor.process(analysis.relation(), sessionContext)) {
+                throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                    "User \"%s\" is not authorized to execute statement \"%s\"",
+                    sessionContext.userName(), analysis));
+            }
+            return true;
+        }
+
+        @Override
+        protected Boolean visitCreateUserStatement(CreateUserAnalyzedStatement analysis,
+                                                   SessionContext sessionContext) {
+            if (!isSuperUser(sessionContext.userName())){
+                throw new UnsupportedOperationException(String.format(Locale.ENGLISH, "User \"%s\" is not authorized to execute statement \"%s\"",
+                    sessionContext.userName(), analysis));
+            }
+            return true;
+        }
+
+        @Override
+        protected Boolean visitDropUserStatement(DropUserAnalyzedStatement analysis,
+                                                 SessionContext sessionContext) {
+            if (!isSuperUser(sessionContext.userName())){
+                throw new UnsupportedOperationException(String.format(Locale.ENGLISH, "User \"%s\" is not authorized to execute statement \"%s\"",
+                    sessionContext.userName(), analysis));
+            }
+            return true;
+        }
+    }
+
+    private class SelectStatementPermissionVisitor extends AnalyzedRelationVisitor<SessionContext, Boolean> {
+
+        @Override
+        protected Boolean visitAnalyzedRelation(AnalyzedRelation relation, SessionContext context) {
+            return true;
+        }
+
+        @Override
+        public Boolean visitQueriedTable(QueriedTable table, SessionContext context) {
+            return !(table.tableRelation().tableInfo().ident().name().equals("users") &&
```

Comment:
Could just do equals on the ident using the SysUsersTableInfo.IDENT constant. 

Also - by reading the documentation it wasn't clear to me that the sys.users table can also only be read by a super user - the first paragraph is phrased in a way that sounds like only CREATE / DROP are special

Furthermore, this logic here won't work for any subqueries or joins. I'm also not sure if this is the right way to have permissions.
I'd have expected this to be the other way around . E.g. a table knows which permissions/roles are required - not that the UserManger knows about every table